### ISSUE TITLE
Windows ramplot not valid

### ISSUE THREAD

**Issue by tuinboontje** [Chia-Network/bladebit#434]
I have an issue with ram plot. None of the c1 through c4 are valid. c5 it just craps out and hangs will not even plot. the generated log for the plots do not mention any error of any kind thus presumend valid/succeded.
Can't seem to find any info on this problem.
Dual Xeons with 512 GB of Ram so ram shortage can be ruled out. Using the bladebit command line it just says sytem mem and closes. Any help/pointers are much appreciated.

**Comment by tuinboontje** [comment]
WHY does it still make a temp file on the drive when plotting in ram?

**Comment by LeroyINC** [comment]
it writes to the final plot file during the plotting process. so the final write time takes less. as it can commit some of it as it goes.

**Comment by tuinboontje** [comment]
Any ideas regarding the crashing of higher then c1 plots, and being invalid even though the log says it is completed?

**Comment by tuinboontje** [comment]
No one I've got one. It's just poorly poorly optimized read somewhere that bladebit does not like 2 numa nodes. Did 1 it worked at first, but still crashed after 2 invalid plots. All ramplots are invalid. C0 through C5. Even diskplot: cheap nvme read ±400 write 750MB 121 min. Better nvme ± 700  read ±1.1 to 1.4GB a second 62 min so that scales fairly decent. On to a datacenter stripe array that reads 5.5 GB a second and writes around 6GB (they are sustained write optimized) it takes 76 minutes. There is absolutely no rime nor reason for these times.  Get the gnarly feeling that it is mostly AMD optimized. Could be wrong but other diskplotters with similar read n write that are AMD based hit sub to low 20 min. Some even faster....

**Comment by harold-b** [comment]
> No one I've got one. It's just poorly poorly optimized read somewhere that bladebit does not like 2 numa nodes.

This is incorrect. bladebit's `ramplot` I believe is the only plotter explicitly coded w/ NUMA in mind. You might have to choose the best settings for high memory throughput on your BIOS. `ramplot` has not changed since its first versions w/ users running them for months at a time with no crashes, so its the most stable out of the 3 variants.

The only things that changed in the latest v3 were minor things to accommodate for compressed plotting. There could potentially be a bug there in phase 3, but we never encountered any during testing. If you can provide plot id's and compression levels for the plots you created that were invalid I could reproduce them locally to see if I encounter them as well. If you have some full logs of ramplot's faulty plots please post a few as well and I can take a look

**Comment by harold-b** [comment]
Are the corrupt ones only in Windows, by the way?

**Comment by tuinboontje** [comment]
Yes sir they are. Where could I find these, do you mean the actual id's of the plots. If so I will post them. As for logs they just crap out at the start at first anfmd aftercreadong about the numa dislike i changed it to no more than 16 threads, thus using  1 numa node. Thats just it the logs state completed succesfully no errors. But the farmer node does not recognise them as valid

**Comment by tuinboontje** [comment]
Ok i think i got the not opening compressed plots sorted. was my bad and not the harvester the decrompression was turned of. I discovered this after reading the debug logs. started a series of 10 C5. 1st C5 did seize up at start and never got to allocating buffers. Deleted it and it suprisingly started with 2nd plot... Got a autoremover so i hope it  will krank out 10 plots, o on sorry 9 the first 1 did choke upon start

**Comment by tuinboontje** [comment]
So it is a hit and a miss. Never makes it past third plot, makes 1 delete the next where it keeps hanging and then it will start the next. It is not set and forget but rather set and forget about it. as if plotting on its own when automated did not take long enough, i now have to keep an eye out for when it crashes, because it will!! and run all night burning electric on a plot that is never going to finish. Maybe code in sort of a counter that exits current plot when phase time equals or exceeds number x and then start with next in the batch. It's a fairly safe assumption when a phase or rather a subset lets say propagating, sorting or computing fx in the phase is not completed after say 1800 seconds it most likely never will

**Comment by tuinboontje** [comment]
Ok it plowed through 7 plots un interrupted. Thats a plus. Now i can realish into gpu plotting, as my riser and bracket for my server finaly got delivered in  the mail this afternoon. But still going to check out if i can add a boundry of some sorts to define scope of "normal" operating parameters. Still think that 30 minutes for allocating buffers and resources will more then suffice for atleast ram and cudaplotting. No need to reinvent the wheel, that is well beyond the scope of my coding capabilities and lets be honest it is a very well designed wheel already. Even the temp cleanup is already arranged in the code i think, because it litteraly is done at the end.

**Comment by tuinboontje** [comment]
After careful reading your reply stating it is the most stable out of the 3( so you do know there are issues with the 3), is like spinning 3 spindles a b c. A 100mph b 75 mph c 50 mph and stating a is the most stable( the longest), which it is due to higher rpm and the gyroscopic effect. But the end result wil be the same for all 3. Lay on their side.....

**Comment by tuinboontje** [comment]
That said gpu plotter crashed after 1 plot out of 10. After deleting it did finish. On go 25.. crashes after 16. Luckily I caught it and after checking the logs from the mover I concluded that it hanged on allocating resources for over 45 of should I say only 45 minutes.
